engine/src/test/java/org/kigalisim/validate/LiveTests.java

```diff
@@ -340,7 +340,174 @@ public void testBasicUnitsConvert() throws IOException {
         "Manufacture units should be kg");
   }
 
-  /*
+  /**
+   * Test basic_replace with retire only - to isolate the issue.
+   */
+  @Test
+  public void testBasicReplaceRetireOnly() throws IOException {
+    // Load and parse the QTA file
+    String qtaPath = "/tmp/test_basic_replace_retire_only.qta";
+    ParsedProgram program = KigaliSimFacade.parseAndInterpret(qtaPath);
+    assertNotNull(program, "Program should not be null");
+
+    // Get the scenario
+    String scenarioName = "Sim";
+    ParsedScenario scenario = program.getScenario(scenarioName);
+    assertNotNull(scenario, "Scenario should exist");
+
+    // Get start and end years from scenario
+    int startYear = scenario.getStartYear();
+    int endYear = scenario.getEndYear();
+
+    // Create engine and machine
+    Engine engine = new SingleThreadEngine(startYear, endYear);
+    PushDownMachine machine = new SingleThreadPushDownMachine(engine);
+
+    // Execute the default policy
+    ParsedPolicy defaultPolicy = program.getPolicy("default");
+    assertNotNull(defaultPolicy, "Default policy should exist");
+    executePolicy(defaultPolicy, machine);
+
+    // Execute other policies in the scenario
+    for (String policyName : scenario.getPolicies()) {
+      ParsedPolicy policy = program.getPolicy(policyName);
+      assertNotNull(policy, "Policy " + policyName + " should exist");
+      executePolicy(policy, machine);
+    }
+
+    // Move to year 10 and check results (like the JavaScript test)
+    for (int year = 2; year <= 10; year++) {
+      engine.incrementYear();
```

Comment:
This is not enough to actually run the simulation from year to year. We should transition `LiveTests` as a whole to use `KigaliSimFacade`